tests/integration/components/self-clear-cache-test.js

```diff
@@ -0,0 +1,93 @@
+import { module, test } from 'qunit';
+import { setupRenderingTest } from 'ember-qunit';
+import { click, render } from '@ember/test-helpers';
+import { hbs } from 'ember-cli-htmlbars';
+
+module('Unit | Component | self-clear-cache', (hooks) => {
+  setupRenderingTest(hooks);
+
+  test('should show last time user cleared the cache', async function (assert) {
+    assert.expect(1);
+
+    // set the outer context '23 March 1:23 pm IST'
+    const time = '23 March 1:23 pm IST';
+    this.set('lastTime', time);
+
+    await render(hbs`<SelfClearCache @time={{this.lastTime}} />`);
+
+    assert.equal(
+      assert.dom('[data-test-last-time]').hasText(`Last Requested : ${time}`),
+      true,
+      'Last time it was requested at 23 March 1:23 pm IST'
+    );
+  });
+
+  // Clear cache button should get disabled on clicking it
+  test('Clear cache button should get disabled on clicking it', async function (assert) {
+    assert.expect(2);
+
+    // set the outer context '23 March 1:23 pm IST'
+    this.set('lastTime', '23 March 1:23 pm IST');
+
+    await render(hbs`<SelfClearCache @time={{this.lastTime}} />`);
+
+    const btn = assert.dom('[data-test-clear-cache-btn]');
```

Comment:
Not installed this add-on yet? 🤔 
https://github.com/simplabs/ember-test-selectors